redis/src/test/scala/zio/redis/OutputSpec.scala

```diff
@@ -498,7 +498,182 @@ object OutputSpec extends BaseSpec {
             )
           )
           Task(XReadOutput.unsafeDecode(input)).either.map(assert(_)(isLeft(isSubtype[ProtocolError](anything))))
-        }
+        },
+        suite("XInfo stream commands output")(
+          testM("stream extract valid value") {
+            val input = RespValue.array(
+              RespValue.bulkString("length"),
+              RespValue.Integer(1),
+              RespValue.bulkString("radix-tree-keys"),
+              RespValue.Integer(2),
+              RespValue.bulkString("radix-tree-nodes"),
+              RespValue.Integer(3),
+              RespValue.bulkString("groups"),
+              RespValue.Integer(2),
+              RespValue.bulkString("last-generated-id"),
+              RespValue.bulkString("1-111"),
+              RespValue.bulkString("first-entry"),
+              RespValue.array(
+                RespValue.bulkString("id"),
+                RespValue.bulkString("1-0"),
+                RespValue.array(
+                  RespValue.bulkString("key1"),
+                  RespValue.bulkString("value1")
+                )
+              ),
+              RespValue.bulkString("last-entry"),
+              RespValue.array(
+                RespValue.bulkString("id"),
+                RespValue.bulkString("1-0"),
+                RespValue.array(
+                  RespValue.bulkString("key2"),
+                  RespValue.bulkString("value2")
+                )
+              )
+            )
+            Task(StreamInfoOutput.unsafeDecode(input)).map(x =>
+              assert(x.lastGeneratedId)(equalTo(Some("1-111"))) &&
+                assert(x.firstEntry)(equalTo(Some(StreamEntry("1-0", Map("key1" -> "value1"))))) &&
+                assert(x.lastEntry)(equalTo(Some(StreamEntry("1-0", Map("key2" -> "value2")))))
+            )
+          },
+          testM("has undefined when message mis fields") {
+            val input = RespValue.array(
+              RespValue.bulkString("length"),
+              RespValue.Integer(1),
+              RespValue.bulkString("invalid-field"),
+              RespValue.Integer(2),
+              RespValue.bulkString("radix-tree-nodes"),
+              RespValue.Integer(3)
+            )
+            Task(StreamInfoOutput.unsafeDecode(input)).map { f =>
+              assert(f.undefined)(
+                equalTo(
+                  Map("invalid-field" -> RespValue.Integer.apply(2))
+                )
+              )
+            }
+          },
+          testM("success when message mis value") {
```

Comment:
I suggest that you change the description to `extract valid value with missing fields`.